src/coreclr/jit/ssabuilder.cpp

```diff
@@ -852,68 +853,26 @@ void SsaBuilder::RenameDef(GenTree* defNode, BasicBlock* block)
     }
     else
     {
-        isLocal = defNode->DefinesLocal(m_pCompiler, &lclNode, &isFullDef);
+        isLocal = defNode->DefinesLocal(m_pCompiler, &lclNode, &isFullDef, &offset, &storeSize);
     }
 #else
-    bool isLocal = defNode->DefinesLocal(m_pCompiler, &lclNode, &isFullDef);
-#endif
-=======
-    ssize_t              offset    = 0;
-    unsigned             storeSize = 0;
     bool                 isLocal   = defNode->DefinesLocal(m_pCompiler, &lclNode, &isFullDef, &offset, &storeSize);
->>>>>>> d7d154d7e25b5a4472b75c963b0a73dc23f5fb9b
+#endif
 
     if (isLocal)
     {
+#ifndef TARGET_WASM
         // This should have been marked as definition.
         assert(((lclNode->gtFlags & GTF_VAR_DEF) != 0) && (((lclNode->gtFlags & GTF_VAR_USEASG) != 0) == !isFullDef));
+#endif // TARGET_WASM
```

Comment:
This assert should (must) hold for LIR as well. If there are cases where it doesn't they should be fixed.

Reply:
ok, I hit this with a `0x0000027d84d9d4c8 135: [[GT_STORE_LCL_FLD, TYP_STRUCT]` where the flags are `8000 000D` i.e.
`GTF_VAR_DEF` (good) but not `GTF_VAR_USEASG` when `isFullDef  == false`.  Do you want the full dump?

Reply:
I have commited a change for this, basically ive just taken the logic that is in the assert and used it to set `GTF_VAR_USEASG`.